purchase_requisition_from_rfq/models/purchase_requisition.py

```diff
@@ -0,0 +1,83 @@
+    )
+
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form',
+                        toolbar=False, submenu=False):
+        res = super(PurchaseRequisition, self).fields_view_get(
+            view_id=view_id, view_type=view_type,
+            toolbar=toolbar, submenu=submenu)
+        if res and view_type in ['tree', 'form']:
+            orders = self.env['purchase.order'].browse(
+                self._context.get('active_ids', []))
+            order_lines = self.env['purchase.order.line']
+            te = orders.mapped('requisition_id')
+            if any(order.state not in ('draft', 'send', 'to approve')
+                   for order in orders):
+                raise ValidationError(_('The requisition cannot be processed '
+                                        'because the order is confirmed!'))
+            if te:
+                raise ValidationError(_(
+                    'Order: %s, already agreement created!') % ', '.join(
+                    orders.filtered(lambda l: l.requisition_id).mapped('name')
+                    ))
+            if len(orders) > 1:
+                order_origin = orders[0]
+                if any(len(order.order_line) != len(order_origin.order_line)
+                       for order in orders):
+                    raise ValidationError(_('The requisition cannot be processed '
+                                            'because the order are difference!'))
+                for order in orders:
+                    for line in order.order_line:
+                        origin = order_lines.search(
+                            [('order_id', '=', order_origin.id),
+                             ('product_id', '=', line.product_id.id)])
+                        if line.product_qty != origin.product_qty:
+                            raise ValidationError(_('The requisition cannot be '
+                                                    'processed because the order '
+                                                    'are difference!'))
+        return res
+
+    @api.multi
+    def _get_line(self):
+        orders = self.env['purchase.order'].browse(
+            self._context.get('active_ids', []))
+        if len(orders) > 1:
+            order = orders[0]
+        else:
+            order = orders
+        data_list = []
+        for line in order.order_line:
+            data_list.append((0, 0, {
+                'product_id': line.product_id.id,
+                'product_uom_id': line.product_uom.id,
+                'product_qty': line.product_qty,
+                'price_unit': 0.0,
+            }))
+        return data_list
+
+    @api.multi
+    def action_in_progress(self):
+        res = super(PurchaseRequisition, self).action_in_progress()
+        self.origin = ', '.join(self.purchase_ids.mapped('name'))
+        return res
+
+    @api.multi
+    def write(self, vals):
+        for rec in self:
+            super(PurchaseRequisition, rec).write(vals)
```

Comment:
What is this for?

Reply:
Better to use hook, see sample here.
https://github.com/OCA/account-invoicing/blob/12.0/account_menu_invoice_refund/hooks.py

Reply:
I have this code because many RFQs can create a Agreement.

Reply:
> I have this code because many RFQs can create an Agreement.

Then what? Please elaborate your explaination.